ICAFIX/scripts/run_icasso.m

```diff
@@ -0,0 +1,247 @@
+  figH = findall(0,'type','figure');
+  figH = figH(~contains({figH.Name},'centrotypes'));% the centrotypes figure isn't useful
+  for iF = 1:numel(figH)
+    figFile = sprintf('%s/%s.fig',...
+      outDir,strrep(strrep(strrep(figH(iF).Name,' ','_'),':',''),'Icasso','icasso'));
+    try
+      savefig(figH(iF),figFile,'compact');
+    catch 
+      warning('Could not save icasso figures.')
+    end
+  end
+  close all;
+end
+
+%% set component sign and sort by variance explained
+% set sign
+maxSfinal = max(S_final);
+absminSfinal = abs(min(S_final));
+pos = maxSfinal > absminSfinal;
+neg = maxSfinal < absminSfinal;
+signAll = sign(pos + neg * -1);
+S_final = S_final .* signAll;
+A_final = A_final .* signAll;
+
+% sort by variance explained
+tVar = var(A_final);
+tICAPercentVariances = tVar / sum(tVar) * 100;
+[tICAPercentVariances,Is] = sort(tICAPercentVariances,'descend');
+S_final = S_final(:,Is);
+A_final = A_final(:,Is);
+iq = iq(Is);
+
+W_final = pinv(A_final);
+
+%% calculate single-regression z-stats
+NODEtsnorm = normalise(A_final);% z-score
+pN = pinv(NODEtsnorm); dpN = diag(pN * pN')';
+dof = size(NODEtsnorm,1) - size(NODEtsnorm,2) - 1;
+residuals = demean(vnts, 2) - S_final * NODEtsnorm';
+t = double(S_final ./ sqrt(sum(residuals .^ 2, 2) * dpN / dof));
+Z = zeros(size(t));
+Z(t > 0) = min(-norminv(tcdf(-t(t > 0), dof)),  38.5);
+Z(t < 0) = max( norminv(tcdf( t(t < 0), dof)), -38.5);
+Z(isnan(Z)) = 0;
+
+%% save icasso outputs to match melodic
+% reshape betas (S_final) and z-scores back to 4-d
+mtxDim = [mtxDim(1) Dim];
+volDim = [volDim(1:3) Dim];
+S_final4d = reshape2d4d(S_final,volDim,brainMask,mtxDim);
+Z4d = reshape2d4d(Z,volDim,brainMask,mtxDim);
+
+% save betas and z-score as 4-d nifti volumes
+hdr = niftiinfo(vntsFile);
+hdr.ImageSize(end) = Dim;
+niftiwrite(single(S_final4d),sprintf('%s/melodic_oIC',outDir),hdr,'Compressed',true)
+niftiwrite(single(Z4d),sprintf('%s/melodic_IC',outDir),hdr,'Compressed',true)
+
+% save mixing matrices as tab-delimited text
+dlmwrite(sprintf('%s/melodic_mix',outDir),A_final,'\t');
+dlmwrite(sprintf('%s/melodic_unmix',outDir),W_final,'\t');
+
+% save variance explained as tab-delimited text
+ICstats = [tICAPercentVariances tICAPercentVariances * totVariance]; 
+dlmwrite([outDir '/melodic_ICstats'],ICstats, 'delimiter', '\t');
+
+% copy brainmask into melodic dir
+copyfile(brainMaskFile,[outDir '/mask.nii.gz'])
+
+% save pcaD and pcaE? Looks like FIX doesn't need them
+
+%% calculate FTmix spectra
+ts = struct();
+ts.Nsubjects = 1;
+ts.ts = A_final;
+[ts.NtimepointsPerSubject,ts.Nnodes] = size(ts.ts);
+ts_spectra = nets_spectra_sp(ts);
+
+% save A_final and spectra as tab-delimited text
+dlmwrite([outDir '/melodic_Tmodes'], ts.ts, 'delimiter', '\t'); 
+dlmwrite([outDir '/melodic_FTmix'], ts_spectra, 'delimiter', '\t');
+
+fid = fopen([outDir '/components.txt'],'w');
+fprintf(fid,'%i: Signal\n',1:size(S_final,2));fclose(fid);
+[~,~] = unix(['wb_command -cifti-create-scalar-series ' ...
```

Comment:
Need to do the same for the timeseries, right?  PostFix may also already take care of both of these.